src/test/components/elements/inputs/RegistrationDobInput.test.tsx

```diff
@@ -0,0 +1,168 @@
+    setupTest("/register/student");
+    const dateOfBirthInputs = ["day", "month", "year"].map((unit) =>
+      screen.getByRole("combobox", {
+        name: new RegExp(`${unit} of birth`, "i"),
+      })
+    );
+    dateOfBirthInputs.forEach((input) => expect(input).toBeInTheDocument());
+    const over13Checkbox = screen.getByRole("checkbox", {
+      name: /i am at least 13 years old/i,
+    });
+    expect(over13Checkbox).toBeInTheDocument();
+  });
+
+  it("renders correctly for teacher registration", () => {
+    setupTest("/register/teacher");
+    const dateOfBirthInputs = ["day", "month", "year"].map((unit) =>
+      screen.getByRole("combobox", {
+        name: new RegExp(`${unit} of birth`, "i"),
+      })
+    );
+    dateOfBirthInputs.forEach((input) => expect(input).toBeInTheDocument());
+    const over13Checkbox = screen.queryByRole("checkbox", {
+      name: /i am at least 13 years old/i,
+    });
+    expect(over13Checkbox).not.toBeInTheDocument();
+  });
+
+  it("changes the user details when the date of birth dropdowns are changed", async () => {
+    setupTest("/register/student");
+    const dateOfBirthInputs = ["day", "month", "year"].map((unit) =>
+      screen.getByRole("combobox", {
+        name: new RegExp(`${unit} of birth`, "i"),
+      })
+    );
+    fireEvent.change(dateOfBirthInputs[0], { target: { value: "15" } });
+    fireEvent.change(dateOfBirthInputs[1], { target: { value: "3" } });
+    fireEvent.change(dateOfBirthInputs[2], { target: { value: "2006" } });
+    const expectedDate = new Date("2006-03-15");
+    expect(mockSetUserToUpdate).toHaveBeenCalledTimes(3);
+    expect(mockSetUserToUpdate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        dateOfBirth: expectedDate,
+      })
+    );
+    expect(mockSetDobOver13CheckboxChecked).toHaveBeenCalledTimes(3);
+    expect(mockSetDobOver13CheckboxChecked).toHaveBeenCalledWith(false);
+  });
+
+  it("updates the over-13 value if checkbox is checked", async () => {
+    setupTest("/register/student");
+    const over13Checkbox = screen.getByRole("checkbox", {
+      name: /i am at least 13 years old/i,
+    });
+    fireEvent.click(over13Checkbox);
+    expect(mockSetDobOver13CheckboxChecked).toHaveBeenCalledTimes(1);
+    expect(mockSetDobOver13CheckboxChecked).toHaveBeenCalledWith(true);
+  });
+
+  it("if a valid date of birth is selected, the over 13 checkbox is disabled", async () => {
+    setupTest("/register/student", {
+      userToUpdate: {
+        ...mockUserToUpdate,
+        dateOfBirth: new Date("2006-03-15"),
+      },
+      dobOver13CheckboxChecked: true,
+    });
+    const over13Checkbox = screen.getByRole("checkbox", {
+      name: /i am at least 13 years old/i,
+    });
+    expect(over13Checkbox).toBeDisabled();
+    const dobError = screen.queryByText(/please enter a valid date of birth/i);
+    expect(dobError).toBeNull();
+  });
+
+  it("if an invalid date of birth is selected, an error is shown", async () => {
```

Comment:
The name of this test could be a bit more concise. Rather than an invalid date, you could mention that if the date of birth of the user is less than 13yo, an error is shown.